src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -2213,18 +2213,615 @@ class C
+
+                    public partial int this[int x, int y, int z = 1] { get; set; }
+                    public partial int this[int x, int y, int z = 1] { get => z; set { } }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (4,40): warning CS1066: The default value specified for parameter 'y' will have no effect because it applies to a member that is used in contexts that do not allow optional arguments
+                //     public partial int this[int x, int y = 1] { get => y; set { } }
+                Diagnostic(ErrorCode.WRN_DefaultValueForUnconsumedLocation, "y").WithArguments("y").WithLocation(4, 40),
+                // (7,47): warning CS1066: The default value specified for parameter 'z' will have no effect because it applies to a member that is used in contexts that do not allow optional arguments
+                //     public partial int this[int x, int y, int z = 1] { get => z; set { } }
+                Diagnostic(ErrorCode.WRN_DefaultValueForUnconsumedLocation, "z").WithArguments("z").WithLocation(7, 47));
+        }
+
+        [Fact]
+        public void OptionalParameters_AllParametersAreOptional()
+        {
+            // An indexer access needs at least one argument in order to be valid
+            var source = """
+                partial class C
+                {
+                    void M()
+                    {
+                        _ = this[];
+                        _ = this[1];
+                    }
+
+                    public partial int this[int x = 1, int y = 2] { get; set; }
+                    public partial int this[int x = 1, int y = 2] { get => y; set { } }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (5,18): error CS0443: Syntax error; value expected
+                //         _ = this[];
+                Diagnostic(ErrorCode.ERR_ValueExpected, "]").WithLocation(5, 18),
+                // (10,33): warning CS1066: The default value specified for parameter 'x' will have no effect because it applies to a member that is used in contexts that do not allow optional arguments
+                //     public partial int this[int x = 1, int y = 2] { get => y; set { } }
+                Diagnostic(ErrorCode.WRN_DefaultValueForUnconsumedLocation, "x").WithArguments("x").WithLocation(10, 33),
+                // (10,44): warning CS1066: The default value specified for parameter 'y' will have no effect because it applies to a member that is used in contexts that do not allow optional arguments
+                //     public partial int this[int x = 1, int y = 2] { get => y; set { } }
+                Diagnostic(ErrorCode.WRN_DefaultValueForUnconsumedLocation, "y").WithArguments("y").WithLocation(10, 44));
+        }
```

Comment:
Let's also test differing default values: one part has `1`, the other has `2`, for example. And verify behavior runtime for this scenario if it compiles.